test/extended/pods/pdb.go

```diff
@@ -0,0 +1,182 @@
+
+	g.It("all platform components should have PDBs configured", func() {
+		kubeClient := oc.AdminKubeClient()
+
+		deployments, err := kubeClient.AppsV1().Deployments("").List(context.Background(), metav1.ListOptions{})
+		if err != nil {
+			e2e.Failf("unable to list deployments: %v", err)
+		}
+		statefulsets, err := kubeClient.AppsV1().StatefulSets("").List(context.Background(), metav1.ListOptions{})
+		if err != nil {
+			e2e.Failf("unable to list statefulsets: %v", err)
+		}
+		// ovn-kube, openshift-controller-manager are running as DS.
+		daemonsets, err := kubeClient.AppsV1().DaemonSets("").List(context.Background(), metav1.ListOptions{})
+		if err != nil {
+			e2e.Failf("unable to list daemonsets: %v", err)
+		}
+
+		var items []runtime.Object
+		for i := range deployments.Items {
+			items = append(items, &deployments.Items[i])
+		}
+		for i := range statefulsets.Items {
+			items = append(items, &statefulsets.Items[i])
+		}
+		for i := range daemonsets.Items {
+			items = append(items, &daemonsets.Items[i])
+		}
+		// iterate over the references to find valid images
+		infra, err := oc.AdminConfigClient().ConfigV1().Infrastructures().Get(context.Background(), "cluster", metav1.GetOptions{})
+		if err != nil {
+			e2e.Failf("unable to get infrastructure: %v", err)
+		}
+		var inHAMode bool
+		if infra.Status.InfrastructureTopology == oapi.HighlyAvailableTopologyMode {
+			inHAMode = true
+		}
+		//controlPlaneStaticPodNames := sets.NewString("etcd", "kube-apiserver", "kube-controller-manager",
+		//	"openshift-kube-scheduler")
+		controlPlaneStaticPodNames, err := getAllControlPlaneStaticPods(kubeClient)
+		if err != nil {
+			e2e.Failf("unable to get names of static pods: %v", err)
+		}
+		// knownBugs := map[string]string{} // Create bugs in future
+		pdbMissingWorkloads := []string{}
+		pdbMisconfiguredWorkloads := []string{}
+		operatorsHavingPDBS := []string{}
+		for _, workload := range items {
+			var meta *metav1.ObjectMeta
+			var selectorLabels *metav1.LabelSelector
+			var replicas int32
+			switch t := workload.(type) {
+			case *appsv1.Deployment:
+				t.Kind = "Deployment"
+				meta = &t.ObjectMeta
+				selectorLabels = t.Spec.Selector
+				replicas = t.Status.Replicas
+			case *appsv1.StatefulSet:
+				t.Kind = "StatefulSet"
+				meta = &t.ObjectMeta
+				selectorLabels = t.Spec.Selector
+				replicas = t.Status.Replicas
+			case *appsv1.DaemonSet:
+				t.Kind = "DaemonSet"
+				meta = &t.ObjectMeta
+				selectorLabels = t.Spec.Selector
+				replicas = t.Status.DesiredNumberScheduled
+			default:
+				panic("not an object")
+			}
+			if !strings.Contains(meta.Namespace, "openshift-") {
+				// Not every component is following the convention of -operator at the end, there are some namespaces
+				// like openshift-monitoring and openshift-network-diagnostics and some of them run on worker nodes
+				// as well. So, need to figure out how to exclude operator components and components running on worker node
```

Comment:
I don't understand the need to exclude operators from this.  If we have an operator with `replicas > 1`, can't we apply the same PDB considerations to it as well?

Reply:
The goal here is to get control plane components most of which run on master nodes.

Reply:
OpenShift-core operators will also run on the control-plane machines.  If any of them have `replicas > 1` (maybe none of them), why wouldn't we want them covered by PDBs?